dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -111,25 +127,83 @@ const getSizes = (
+		if (breakpoint >= breakpoints.wide) return 1020;
+		if (breakpoint >= breakpoints.leftCol) return 940;
+		if (breakpoint >= breakpoints.tablet) return 700;
+		if (breakpoint >= breakpoints.phablet) return 700;
+		return breakpoint;
 	}
 
 	switch (role) {
 		case 'inline':
-			return `(min-width: ${breakpoints.phablet}px) 620px, 100vw`;
+			if (
+				breakpoint >= breakpoints.tablet &&
+				breakpoint < breakpoints.desktop
+			)
+				return 680;
+			if (breakpoint >= breakpoints.phablet) return 620;
+			return breakpoint;
 		case 'halfWidth':
-			return `(min-width: ${breakpoints.phablet}px) 300px, 50vw`;
+			if (breakpoint >= breakpoints.phablet) return 300;
+			return Math.round(breakpoint / 2);
 		case 'thumbnail':
-			return '140px';
+			return 140;
 		case 'immersive':
-			return `(min-width: ${breakpoints.wide}px) 1300px, 100vw`;
+			if (breakpoint >= breakpoints.wide) return 1300;
+			return breakpoint;
 		case 'supporting':
-			return `(min-width: ${breakpoints.wide}px) 380px, 300px`;
+			if (breakpoint >= breakpoints.wide) return 380;
+			if (breakpoint >= breakpoints.tablet) return 300;
+			return breakpoint;
 		case 'showcase':
-			return `(min-width: ${breakpoints.wide}px) 860px, (min-width: ${breakpoints.leftCol}px) 780px, (min-width: ${breakpoints.phablet}px) 620px, 100vw`;
+			if (breakpoint >= breakpoints.wide) return 860;
+			if (breakpoint >= breakpoints.leftCol) return 780;
+			if (breakpoint >= breakpoints.phablet) return 620;
+			return breakpoint;
 	}
 };
 
+// Takes a size & sources, and returns a source set with 1 matching image with the desired size
+const getSourceForDesiredWidth = (
+	desiredWidth: Pixel,
+	sources: SrcSetItem[],
+	resolution: ResolutionType,
+) => {
+	// The image sources we're provided use double-widths for HDPI images
+	// We should therefor multiply the width based on if it's HDPI or not
+	// e.g { url: '... ?width=500&dpr=2 ...', width: '1000' }
+	const source = getBestSourceForDesiredWidth(
+		resolution === 'hdpi' ? desiredWidth * 2 : desiredWidth,
+		sources,
+	);
+	return `${source.src} ${source.width}w`;
+};
+
+// Create sourcesets for portrait immersive
+// TODO: In a future PR this system will be updated to solve scaling issues with DPR
+const portraitImmersiveSource = (
+	sources: SrcSetItem[],
+	resolution: ResolutionType,
+) => (
+	<source
+		media={
+			resolution === 'hdpi'
+				? '(orientation: portrait) and (-webkit-min-device-pixel-ratio: 1.25), (orientation: portrait) and (min-resolution: 120dpi)'
+				: '(orientation: portrait)'
+		}
+		// Immersive MainMedia elements fill the height of the viewport, meaning
+		// on mobile devices even though the viewport width is small, we'll need
+		// a larger image to maintain quality. To solve this problem we're using
+		// the viewport height (vh) to calculate width. The value of 167vh
+		// relates to an assumed image ratio of 5:3 which is equal to
+		// 167 (viewport height)  : 100 (viewport width).
+		sizes="167vh"
```

Comment:
This is clever, but ideally we would have a portrait source. Otherwise the majority of the image downloaded isn’t even used.

To investigate in another PR, though.

Reply:
Indeed we could use resizer to cut off sides in this scenario. It could mean, browsers will have to download another image when orientation or window ratio changes, though. Worth investigating!

Also worth noting, the assumption of 5:3 crop, while practically often true at the Guardian, isn’t necessary as we carry real ratio in CAPI.

Reply:
> as we carry real ratio in CAPI

Oh, so we could replace `167` with a value calculated based on `ratio`? That seems like an easy win.

Reply:
Waaaait… Why I can see `aspectRatio` only for the trail (always 5:3) image, but not for Main media which eg. [here](https://www.theguardian.com/artanddesign/2021/nov/23/tom-stoddart-obituary) is a different ratio?! We could calculate from `width` and `height`, but I will ask CAPI why `aspectRatio` isn’t offered for all imagery.